streams/src/main/java/org/apache/kafka/streams/kstream/internals/SessionKeyBinaryConverter.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.kstream.Windowed;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Utility class for the {@link org.apache.kafka.streams.state.SessionStore}.
+ * Used to convert to/from the binary & non-binary representations of the
+ * {@link Windowed} key used in the store.
+ * @param <K>
+ */
+public class SessionKeyBinaryConverter<K> {
```

Comment:
The end timestamp can't be calculated for `SessionWindows` as it is not of a fixed length. We need to store both start and end in this case